CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.15.0 FATAL_ERROR) # Configurable policies: <= CMP0094
+cmake_minimum_required(VERSION 3.17.0 FATAL_ERROR) # Configurable policies: <= CMP0094
```

Comment:
the LGTM service uses a somewhat old cmake, and its analysis of libtorrent depends on it (this is why `master` isn't analyzed right now). I think this is risky, but I can't recall which version LGTM requires

Reply:
@arvidn I also could not find information about this. However, I forced LGTM to run and it succeeded, so it seems to be fine.